stream-helpers/filter.ts

```diff
@@ -0,0 +1,50 @@
+import type { Operation, Stream } from "effection";
+
+/**
+ * Filters items from the stream based on a predicate function.
+ *
+ * @param predicate - The function to test each item
+ * @returns A stream transformer that only emits items that pass the predicate
+ *
+ * @example
+ * ```typescript
+ * import { filter } from "@effectionx/stream-helpers";
+ * import { run, each } from "effection";
+ *
+ * await run(function* () {
+ *   const stream = filter((x: number) => x > 5)(sourceStream);
```

Comment:
Let's use multi-step currying as in other examples. It re-enforces at every point why the api is written this way.

Reply:
Sorry, what is the right format that we should be using?